src/mgmt/srv.c

```diff
@@ -406,6 +504,29 @@ process_drpc_request(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 		drpc_resp->body.data = body;
 		D_FREE(join_resp);
 		break;
+	case DRPC_METHOD_MGMT_CREATE_POOL:
+		D_ALLOC_PTR(create_pool_resp);
+		if (create_pool_resp == NULL) {
+			drpc_resp->status = DRPC__STATUS__FAILURE;
+			D_ERROR("Failed to allocate daos response ref\n");
+			break;
+		}
+		process_create_pool_request(drpc_req, create_pool_resp);
+		len = mgmt__create_pool_resp__get_packed_size(create_pool_resp);
+		D_ALLOC(body, len);
+		if (body == NULL) {
+			drpc_resp->status = DRPC__STATUS__FAILURE;
+			D_ERROR("Failed to allocate drpc response body\n");
+			D_FREE(create_pool_resp);
+			break;
+		}
+		mgmt__create_pool_resp__pack(create_pool_resp, body);
+		drpc_resp->body.len = len;
+		drpc_resp->body.data = body;
+		D_FREE(create_pool_resp->svcreps);
+		D_FREE(create_pool_resp->uuid);
```

Comment:
Good. Do we need to do the same before line 520?

By the way, this reminds me that the GetAttachInfo case (my code) made a mistake leaking the memory in the resp. I'll submit a patch to fix that case.

Reply:
It doesn't look like this issue (i.e., freeing svcresp and uuid in the "if (body == NULL)" branch) has been resolved, does it...